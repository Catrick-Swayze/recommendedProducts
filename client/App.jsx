import React from 'react';
import List from './List.jsx';

class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      items: [],
    }
  }

  render() {
    return (
      <h4>More to consider</h4>
      <div id="recommended-items">
        <List listItems={this.state.items}/>
      </div>

    );
  }
}

export default App;
