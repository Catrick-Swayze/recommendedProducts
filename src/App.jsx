import React, { useState, useEffect } from 'react';
import List from './List.jsx';
import axios from 'axios';

const App = (props) => {

  const [allItems, setAllItems] = useState([]);
  const [shownItems, setShownItems] = useState([]);

  useEffect(() => {
    // when component mounts, and if no items have been populated:
    if (!allItems.length) {
      // 'garden' is a temporary query for now--will likely be changed for later implementation
      axios.get('/products/dept/garden')
        .then(results => {
          console.log('inside get request');
          setAllItems(results.data);
          setShownItems(results.data.slice(0, 6));
        })
        .catch(err => console.log('there was an error', err));
    }
  });

  return (
    <div>
      <h4>More to consider</h4>
      <div id="recommended-items">
        <List listItems={shownItems}/>
      </div>
    </div>
  );
}

export default App;
