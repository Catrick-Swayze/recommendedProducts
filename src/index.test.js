import React from 'react';
import renderer from 'react-test-renderer';
import App from './App.jsx';

test('test suite is functional', () => {
  expect(true).toBe(true);
});

test('should render at least one ListItem component', () => {
  const component = renderer.create(
    <App />
  );
  let app = component.toJSON();
  expect(app).toMatchSnapshot();
});

/*
test('should render at least one ListItem component', () => {});
*/
