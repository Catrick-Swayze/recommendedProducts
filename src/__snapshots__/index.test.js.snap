// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render at least one ListItem component 1`] = `
<div>
  <h4
    className="sc-bdnylx cvwJVX"
  >
    More to consider
  </h4>
  <div>
    <ul
      className="sc-hKFyIo jRfBt list"
    >
      <li
        className="sc-eCApGN EpVoV list-item"
        onClick={[Function]}
      >
        <a
          href="http://fakewebsite.com/products/0"
        >
          <img
            className="sc-jSFkmK oIuXF"
            src="https://twzkraus-fec-images.s3-us-west-1.amazonaws.com/50+images/0.jpg"
          />
          <p
            className="sc-gtssRu bTjxwM item-price"
          >
            31.00
          </p>
          <p
            className="sc-dlnjPT hCTWnp item-title"
          >
            Intelligent Metal Cheese
          </p>
        </a>
      </li>
      <li
        className="sc-eCApGN EpVoV list-item"
        onClick={[Function]}
      >
        <a
          href="http://fakewebsite.com/products/1"
        >
          <img
            className="sc-jSFkmK oIuXF"
            src="https://twzkraus-fec-images.s3-us-west-1.amazonaws.com/50+images/1.jpg"
          />
          <p
            className="sc-gtssRu bTjxwM item-price"
          >
            88.00
          </p>
          <p
            className="sc-dlnjPT hCTWnp item-title"
          >
            Handmade Metal Hat
          </p>
        </a>
      </li>
      <li
        className="sc-eCApGN EpVoV list-item"
        onClick={[Function]}
      >
        <a
          href="http://fakewebsite.com/products/2"
        >
          <img
            className="sc-jSFkmK oIuXF"
            src="https://twzkraus-fec-images.s3-us-west-1.amazonaws.com/50+images/2.jpg"
          />
          <p
            className="sc-gtssRu bTjxwM item-price"
          >
            76.00
          </p>
          <p
            className="sc-dlnjPT hCTWnp item-title"
          >
            Refined Metal Ball
          </p>
        </a>
      </li>
      <li
        className="sc-eCApGN EpVoV list-item"
        onClick={[Function]}
      >
        <a
          href="http://fakewebsite.com/products/3"
        >
          <img
            className="sc-jSFkmK oIuXF"
            src="https://twzkraus-fec-images.s3-us-west-1.amazonaws.com/50+images/3.jpg"
          />
          <p
            className="sc-gtssRu bTjxwM item-price"
          >
            66.00
          </p>
          <p
            className="sc-dlnjPT hCTWnp item-title"
          >
            Practical Frozen Car
          </p>
        </a>
      </li>
      <li
        className="sc-eCApGN EpVoV list-item"
        onClick={[Function]}
      >
        <a
          href="http://fakewebsite.com/products/4"
        >
          <img
            className="sc-jSFkmK oIuXF"
            src="https://twzkraus-fec-images.s3-us-west-1.amazonaws.com/50+images/4.jpg"
          />
          <p
            className="sc-gtssRu bTjxwM item-price"
          >
            54.00
          </p>
          <p
            className="sc-dlnjPT hCTWnp item-title"
          >
            Generic Cotton Salad
          </p>
        </a>
      </li>
      <li
        className="sc-eCApGN EpVoV list-item"
        onClick={[Function]}
      >
        <a
          href="http://fakewebsite.com/products/5"
        >
          <img
            className="sc-jSFkmK oIuXF"
            src="https://twzkraus-fec-images.s3-us-west-1.amazonaws.com/50+images/5.jpg"
          />
          <p
            className="sc-gtssRu bTjxwM item-price"
          >
            5.00
          </p>
          <p
            className="sc-dlnjPT hCTWnp item-title"
          >
            Sleek Soft Car
          </p>
        </a>
      </li>
      <li
        className="sc-eCApGN EpVoV list-item"
        onClick={[Function]}
      >
        <a
          href="http://fakewebsite.com/products/6"
        >
          <img
            className="sc-jSFkmK oIuXF"
            src="https://twzkraus-fec-images.s3-us-west-1.amazonaws.com/50+images/6.jpg"
          />
          <p
            className="sc-gtssRu bTjxwM item-price"
          >
            5.00
          </p>
          <p
            className="sc-dlnjPT hCTWnp item-title"
          >
            Practical Rubber Table
          </p>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
